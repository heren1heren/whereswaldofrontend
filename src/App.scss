.color {
    color: red;
}

.button {
    margin-left: 100px;
}
