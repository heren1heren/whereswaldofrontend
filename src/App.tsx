import { useEffect, useState } from 'react';
import styled from 'styled-components';
import './App.scss';
// how to make fetching an image reusable?
// creating custom hook
const RedButton = styled.button`
  background-color: red;
  margin-left: 50px;
  display: flex;
  justify-content: center;
`;
function useImageUrl() {
  const [imageURL, setImageURL] = useState(null);
  const [isError, setIsError] = useState(false);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    (async () => {
      try {
        const response = await fetch(
          'https://jsonplaceholder.typicode.com/photos',
          {
            mode: 'cors',
          }
        );
        const result = await response.json();
        setImageURL(result[0].url);
      } catch (error) {
        // fetch a default image or display error
        setIsError(true);
        console.log(error);
      } finally {
        // here is how to use finally
        setLoading(false);
      }
    })();
  }, []);
  return { imageURL, loading, isError };
}
const Image = () => {
  const { imageURL, isError, loading } = useImageUrl();

  if (isError) return <p>A network error was encountered</p>;
  if (loading) return <p> Happy loading...</p>;
  return (
    imageURL && (
      <>
        <RedButton className="button" id="button">
          {' '}
          hello
        </RedButton>
        <h1 className="button">An image</h1>

        <img src={imageURL} alt={'placeholder text'} />
      </>
    )
  );
};

export default Image;
