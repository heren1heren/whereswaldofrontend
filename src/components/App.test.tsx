import { render, screen } from '@testing-library/react';
import App from '../router-template';
// need to put the component inside a router to test it
import { MemoryRouter } from 'react-router-dom';
describe('initializing home page test ', () => {
  it('init', () => {
    const component = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
    // get all links
    console.log(screen.getAllByRole('link'));
    expect(screen.getAllByRole('link')).toHaveLength(9);
    // get 6 links inside main page
    // const Main = screen.getBy //
    //*userevent inside the home page component
  });
});
