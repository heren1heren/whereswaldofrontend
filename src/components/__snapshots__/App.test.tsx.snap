// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`initializing home page test  > init 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" d-flex  flex-column justify-content-between vh-100 text-center"
      >
        <h1>
          where is waldo
        </h1>
        <nav
          class="nav justify-content-center  "
        >
          <a
            class="nav-link "
            href="/records"
          >
             
            Record
          </a>
          <a
            class="nav-link "
            href="/"
          >
             
            Home Page
          </a>
        </nav>
        <main
          class=" flex-grow-1 p-lg-4"
        >
           
           
          <p>
            Choose a level:
          </p>
          <div
            class="level-grid h-75"
          >
            <a
              class="level-grid-item"
              href="/map1"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/theGold.jpg"
              />
              The Gold
            </a>
            <a
              class="level-grid-item"
              href="/map2"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/beach.jpg"
              />
              The beach
            </a>
            <a
              class="level-grid-item"
              href="/map3"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/blue.jpg"
              />
              The Blue
            </a>
            <a
              class="level-grid-item"
              href="/map4"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/maze.jpg"
              />
              The Maze
            </a>
            <a
              class="level-grid-item"
              href="/map5"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/snowMountain.jpg"
              />
              The White
            </a>
            <a
              class="level-grid-item"
              href="/map6"
            >
              <img
                alt="where is waldo image"
                class="img-template"
                src="src/assets/space.jpg"
              />
              The Space
            </a>
          </div>
        </main>
        <footer
          class=" text-center bg-black text-white"
        >
           
          Made By
          <a
            class=" ms-md-1  text-decoration-none"
            href="https://github.com/heren1heren/whereswaldofrontend"
          >
            Cookie Turtle
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" d-flex  flex-column justify-content-between vh-100 text-center"
    >
      <h1>
        where is waldo
      </h1>
      <nav
        class="nav justify-content-center  "
      >
        <a
          class="nav-link "
          href="/records"
        >
           
          Record
        </a>
        <a
          class="nav-link "
          href="/"
        >
           
          Home Page
        </a>
      </nav>
      <main
        class=" flex-grow-1 p-lg-4"
      >
         
         
        <p>
          Choose a level:
        </p>
        <div
          class="level-grid h-75"
        >
          <a
            class="level-grid-item"
            href="/map1"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/theGold.jpg"
            />
            The Gold
          </a>
          <a
            class="level-grid-item"
            href="/map2"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/beach.jpg"
            />
            The beach
          </a>
          <a
            class="level-grid-item"
            href="/map3"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/blue.jpg"
            />
            The Blue
          </a>
          <a
            class="level-grid-item"
            href="/map4"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/maze.jpg"
            />
            The Maze
          </a>
          <a
            class="level-grid-item"
            href="/map5"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/snowMountain.jpg"
            />
            The White
          </a>
          <a
            class="level-grid-item"
            href="/map6"
          >
            <img
              alt="where is waldo image"
              class="img-template"
              src="src/assets/space.jpg"
            />
            The Space
          </a>
        </div>
      </main>
      <footer
        class=" text-center bg-black text-white"
      >
         
        Made By
        <a
          class=" ms-md-1  text-decoration-none"
          href="https://github.com/heren1heren/whereswaldofrontend"
        >
          Cookie Turtle
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
